src/com/cft/contactmerge/StreetAddressMatchLogic.java

```diff
@@ -2,13 +2,185 @@
+        put("east", "east");
+        put("w", "west");
+        put("west", "west");
+    }};
+
+    /*******************************************************************************************************************
+     *************************************************** Get Methods ***************************************************
+     *******************************************************************************************************************/
+    public static Hashtable<String,String> getRoadTypes(){
+        return roadTypes;
+    }
+
+    public static Hashtable<String,String> getCompassDirection(){
+        return compassDirection;
+    }
+
+    public static ArrayList<String> getPoBoxTypes(){
+        return poBoxTypes;
+    }
+
     /*******************************************************************************************************************
      ************************************************* Matching Methods ************************************************
      *******************************************************************************************************************/
-    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddress1, ArrayList<String> stAddress2){
-        // ToDo: Fill in code to obtain matching result
+    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddressParts1, ArrayList<String> stAddressParts2){
+        // Precondition: All addressParts must follow the rules from the setNormalizeStreetAddress()
+        String st1Direction = "";
+        String st2Direction = "";
+        String st1RoadType = "";
+        String st2RoadType = "";
+
+        // get return on po box addressses (if either one is a po box)
+        if (isAddressPoBox(stAddressParts1) && isAddressPoBox(stAddressParts2)){
+            // Strip all characters out of address1, and only leave numbers
+            for (int part = 0; part < stAddressParts1.size(); part++){
+                if (!isInteger(stAddressParts1.get(part))){
+                    stAddressParts1.remove(part);
+                    part--;
+                }
+            }
+            // Strip all characters out of address2, and only leave numbers
+            for (int part = 0; part < stAddressParts2.size(); part++) {
+                if (!isInteger(stAddressParts2.get(part))) {
+                    stAddressParts2.remove(part);
+                    part--;
+                }
+            }
+            // if both box addresses match, return yes
+            if (stAddressParts1.equals(stAddressParts2)){
+                return AnswerType.yes;
+            } else {
+                return AnswerType.no;
+            }
+        } else if (isAddressPoBox(stAddressParts1) ^ isAddressPoBox(stAddressParts2)){
+            return AnswerType.no;
+        }
+
+
+        // abstract street direction if the address has one
+        int stDirectionElement = 0;
+        for (String stPart: stAddressParts1){
```

Comment:
Can't we just take the stAddressParts1 and remove all the CompassDirections? Something like

for (String roadType: roadTypes) { stAddressParts1.remove(roadType) }  

It would have to be rewritten to work with a HashTable or put the directions in a List.

Reply:
I can do this... but currently the match is set to yes, if they are the same... and maybe if one is missing. So I'm not sure what the blanket answer should be.